spring-kafka/src/main/java/org/springframework/kafka/core/KafkaStreamsConfigurer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.core;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.processor.StateRestoreListener;
+
+/**
+ * Wrapper class of configurable properties of {@link KafkaStreams}. Note that
+ * {@code setters} of this configurer has precedence over {@link KafkaStreamsCustomizer}s
+ *
+ * @author Nurettin Yilmaz
+ */
+class KafkaStreamsConfigurer {
```

Comment:
I don't see reason in this class. Sounds like `CompositeKafkaStreamsCustomizer` would be better solution.

Reply:
Could you eloborate on this a little bit? I though that separating settable fields like `stateListener`, `stateRestoreListener` and `uncaughtExceptionHandler` and configure logic from `StreamsBuilderFactoryBean` is more readable.